editor/src/components/shared/scoped-variables.ts

```diff
@@ -83,26 +99,79 @@ function getVariablesFromComponent(
   const jsxComponentVariables = variablesInScopeFromEditorState[elementPathString] ?? {}
   return {
     filePath: jsxComponent?.name ?? 'Component',
-    variables: Object.entries(jsxComponentVariables).flatMap(([name, value]) => {
-      const type = getTypeByValue(value)
-      const variable = {
-        name,
-        value,
-        type,
+    variables: generateVariableTypes(jsxComponentVariables),
+  }
+}
+
+function getComponentPropsInScope(
+  topLevelElements: TopLevelElement[],
+  elementPath: ElementPath,
+  jsxMetadata: ElementInstanceMetadataMap,
+): AllVariablesInScope {
+  const jsxComponent = topLevelElements.find((el) => isUtopiaJSXComponent(el)) as UtopiaJSXComponent
```

Comment:
If you define the return value of the inline function, then you don't need to cast the type:
```typescript
const jsxComponent = topLevelElements.find((el): el is UtopiaJSXComponent =>
    isUtopiaJSXComponent(el),
  )
```

But in this case it is even simpler to directly pass in `isUtopiaJSXComponent`
```typescript
const jsxComponent = topLevelElements.find(isUtopiaJSXComponent)
```